flutter_vlc_player/example/lib/controls_overlay.dart

```diff
@@ -6,112 +6,116 @@ class ControlsOverlay extends StatelessWidget {
+                        ),
+                        IconButton(
+                          onPressed: _play,
+                          color: _iconColor,
+                          iconSize: _playButtonIconSize,
+                          icon: Icon(Icons.play_arrow),
+                        ),
+                        IconButton(
+                          onPressed: _seekRelative(_seekStepForward),
+                          color: _iconColor,
+                          iconSize: _seekButtonIconSize,
+                          icon: Icon(Icons.forward_10),
                         ),
-                      ),
-                    );
+                      ],
+                    ),
+                  ),
+                ),
+              );
 
-                  case PlayingState.buffering:
-                  case PlayingState.playing:
-                    return SizedBox.shrink();
+            case PlayingState.buffering:
+            case PlayingState.playing:
+              return GestureDetector(onTap: _pause);
 
-                  case PlayingState.ended:
-                  case PlayingState.error:
-                    return Center(
-                      child: IconButton(
-                        onPressed: () async {
-                          await controller.play();
-                        },
-                        color: Colors.white,
-                        iconSize: 100.0,
-                        icon: Icon(Icons.replay),
-                      ),
-                    );
-                }
-              }
+            case PlayingState.ended:
+            case PlayingState.error:
+              return Center(
+                child: FittedBox(
+                  child: IconButton(
+                    onPressed: _replay,
+                    color: _iconColor,
+                    iconSize: _replayButtonIconSize,
+                    icon: Icon(Icons.replay),
+                  ),
+                ),
+              );
+            default:
               return SizedBox.shrink();
-            },
-          ),
-        ),
-        GestureDetector(
-          onTap: !controller.value.isPlaying
-              ? null
-              : () async {
-                  if (controller.value.isPlaying) {
-                    await controller.pause();
-                  }
-                },
-        ),
-      ],
+          }
+        },
+      ),
     );
   }
+
+  Future<void> _play() {
+    return controller.play();
+  }
+
+  Future<void> _replay() async {
+    await controller.stop();
+    await controller.play();
+  }
+
+  Future<void> _pause() async {
+    if (controller.value.isPlaying) {
+      await controller.pause();
+    }
+  }
+
+  /// Returns a callback which seeks the video relative to current playing time.
+  VoidCallback _seekRelative(Duration seekStep) {
+    return () async {
+      if (controller.value.duration != null) {
+        await controller.seekTo(controller.value.position + seekStep);
+      }
+    };
+  }
```

Comment:
I think it would be a bit more readable to create a method that seeks, instead of creating a callback that seeks. What do you think?  
```suggestion
  void _seekRelative(Duration seekStep) async {
    if (controller.value.duration != null) {
      await controller.seekTo(controller.value.position + seekStep);
    }
  }
```